x-pack/plugins/observability/public/components/custom_threshold/components/alert_details_app_section/alert_details_app_section.tsx

```diff
@@ -172,9 +212,13 @@ export default function AlertDetailsAppSection({
       {ruleParams.criteria.map((criterion, index) => (
         <EuiFlexItem key={`criterion-${index}`}>
           <EuiPanel hasBorder hasShadow={false}>
-            <EuiTitle size="xs">
-              <h4>{criterion.label || 'CUSTOM'} </h4>
-            </EuiTitle>
+            <EuiToolTip content={generateChartTitle(criterion).tooltip}>
```

Comment:
nit: Maybe we can save the return value of `generateChartTitle(criterion)` once and then use `tooltip` and `title` from that variable. The current implementation goes through generating the title twice.

Reply:
updated in 63e1b56351325735006733fe99efb0ee9f742b78